src/redis-cli.c

```diff
@@ -9044,7 +9044,14 @@ int main(int argc, char **argv) {
         if (cliConnect(0) != REDIS_OK) exit(1);
         return evalMode(argc,argv);
     } else {
-        cliConnect(CC_QUIET);
-        return noninteractive(argc,argv);
+        int connected = (cliConnect(CC_QUIET) == REDIS_OK);
+
+        /* Try to serve command even we are not connected. e.g. help command */
+        int retval = noninteractive(argc,argv);
+        if (retval != REDIS_OK) {
```

Comment:
ohh, sorry, i did not think about it back then
I'll change it like this to reduce the number of lines of code (it doesn't matter)
```suggestion
        if (retval != REDIS_OK && !connected) exit(1);
```

Reply:
Sure!